extension/test/integration/bigcommerce/CartRepositoryTest.js

```diff
@@ -141,4 +141,13 @@ describe('BigCommerceCartRepository - integration', () => {
       _subTotals: []
     }])
   })
+
+  it('should catch the correct error message if item not in stock', async () => {
```

Comment:
you technically done really need this fancy mocking to be honest, you can just inject the `BigCommerceCartRepository` into the constructor and rewrite what an update would return. Let's say you want to throw an exception, then you just rewrite the 
```
const error = new Error()
error.code=422
BigCommerceCartRepository.updateItems = async () => { throw new error }
```
Then you can use the addProduct function call and lo and behold it will hit the catch clause. This will make it easier to create tests. You need to also check for all logic, check the message being output by the error. You can skip checking message repository as it's not really needed to be tested here.